tests/robustness/traffic/client.go

```diff
@@ -206,10 +230,38 @@ func (c *RecordingClient) Watch(ctx context.Context, key string, rev int64, with
 		defer close(respCh)
 		for r := range c.client.Watch(ctx, key, ops...) {
 			c.watchMux.Lock()
-			c.watchResponses = append(c.watchResponses, WatchResponse{r, time.Since(c.baseTime)})
+			c.watchResponses = append(c.watchResponses, ToWatchResponse(r, c.baseTime))
 			c.watchMux.Unlock()
 			respCh <- r
 		}
 	}()
 	return respCh
 }
+
+func ToWatchResponse(r clientv3.WatchResponse, baseTime time.Time) WatchResponse {
+	resp := WatchResponse{Time: time.Since(baseTime)}
+	for _, event := range r.Events {
+		resp.Events = append(resp.Events, toWatchEvent(*event))
+	}
+	resp.IsProgressNotify = r.IsProgressNotify()
+	resp.Revision = r.Header.Revision
+	return resp
+}
+
+func toWatchEvent(event clientv3.Event) WatchEvent {
+	var op model.OperationType
+	switch event.Type {
```

Comment:
Sanity check - type is always `PUT` or `DELETE` right?

Reply:
Yes, the type is always PUT and DELETE.  But I agree that we should add a check just in case.

```
default:
    panic(fmt.Sprintf("Unexpected event type: %s", event.Type))
```

Reply:
Done